enterprise/server/remote_execution/cgroup/cgroup.go

```diff
@@ -31,6 +32,127 @@ const (
 	cidPlaceholder = "{{.ContainerID}}"
 )
 
+// Setup configures the cgroup at the given path with the given settings.
+// Any settings for cgroup controllers that aren't enabled are ignored.
+// IO limits are applied to the given block device, if specified.
+func Setup(ctx context.Context, path string, s *scpb.CgroupSettings, blockDevice *block_io.Device) error {
+	m := settingsMap(s, blockDevice)
+	if len(m) == 0 {
+		return nil
+	}
+	enabledControllers, err := ParentEnabledControllers(path)
+	if err != nil {
+		return fmt.Errorf("read enabled controllers: %w", err)
+	}
+	for name, value := range m {
+		controller, _, _ := strings.Cut(name, ".")
+		if !enabledControllers[controller] {
+			log.CtxWarningf(ctx, "Skipping cgroup %q setting for disabled cgroup controller %q", name, controller)
+			continue
+		}
+		settingFilePath := filepath.Join(path, name)
+		if err := os.WriteFile(settingFilePath, []byte(value), 0); err != nil {
+			return fmt.Errorf("write %q to cgroup file %q: %w", value, name, err)
+		}
+	}
+	return nil
+}
+
+// ParentEnabledControllers returns the cgroup controllers that are enabled for
+// the parent cgroup of a given cgroup.
+func ParentEnabledControllers(path string) (map[string]bool, error) {
+	b, err := os.ReadFile(filepath.Join(path, "..", "cgroup.subtree_control"))
+	if err != nil {
+		return nil, err
+	}
+	fields := strings.Fields(string(b))
+	enabled := make(map[string]bool, len(fields))
+	for _, f := range fields {
+		enabled[f] = true
+	}
+	return enabled, nil
+}
+
+func settingsMap(s *scpb.CgroupSettings, blockDevice *block_io.Device) map[string]string {
+	m := map[string]string{}
+	if s == nil {
+		return m
+	}
+	if s.CpuWeight != nil {
+		m["cpu.weight"] = strconv.Itoa(int(s.GetCpuWeight()))
+	}
+	if s.CpuQuotaLimitUsec != nil {
```

Comment:
If just the period is set, this silently ignores it. Maybe we should return an error.